### PR TITLE
Feature/deprecate schema

### PR DESCRIPTION
**Pull request by andym1125**
## Changes

* Added bool isActive to Schema proto
* Implemented DeprecateSchema() in msg_server_schema.go

### New Features *(required)*

DeprecateSchema, which is a method to deactivate a schema on the blockchain. Deprecating an already-deprecated Schema will not do anything and return normally.

### Deprecations *(required)*

N/A

### Enhancements *(optional)*

N/A

## Checklist

* [ ] Unit tests
* [ ] Documentation

## References *(optional)*

Fixes
Connects #355

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

File: x/schema/keeper/msg_server_schema.go [hunk 1]
```diff
@@ -69,6 +69,39 @@ func (k msgServer) CreateSchema(goCtx context.Context, msg *types.MsgCreateSchem
...
+
+	//If already deactivated, do nothing.
+	//Responsibility of caller to check if isActive beforehand
+	if schemaWI.GetIsActive() {
+		schemaWI.IsActive = false
+		k.SetWhatIs(ctx, schemaWI)
+	}
+
+	return &types.MsgDeprecateSchemaResponse{
+		Code:    http.StatusOK,
```
Comment (joshLong145) [inline x/schema/keeper/msg_server_schema.go]: we should change this to be a constant as to remove the http/net dep from this pkg. we can just use the value '200' for now

Comment (andym1125) [reply x/schema/keeper/msg_server_schema.go]: Change made

File: proto/schema/v1/genesis.proto [hunk 2]
```diff
@@ -3,12 +3,15 @@ package sonrio.sonr.schema;
...
 import "schema/v1/params.proto";
+import "schema/v1/what_is.proto";
 // this line is used by starport scaffolding # genesis/proto/import
 
 option go_package = "github.com/sonr-io/sonr/x/schema/types";
 
 // GenesisState defines the schema module's genesis state.
 message GenesisState {
   Params params = 1 [(gogoproto.nullable) = false];
+  repeated WhatIs whatIsList = 2 [(gogoproto.nullable) = false];
```
Comment (mcjcloud) [inline proto/schema/v1/genesis.proto]: In protobuf you need to use underscore_case. This makes sure that protobuf will generate the target code with correct conventions.

File: x/schema/keeper/msg_server_schema.go [hunk 3]
```diff
@@ -69,6 +68,39 @@ func (k msgServer) CreateSchema(goCtx context.Context, msg *types.MsgCreateSchem
...
 func (k msgServer) DeprecateSchema(goCtx context.Context, msg *types.MsgDeprecateSchema) (*types.MsgDeprecateSchemaResponse, error) {
-	// TODO: implement
-	return nil, errors.New("unimplemented")
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	schemas, foundSchemas := k.GetWhatIsFromCreator(ctx, msg.GetCreator())
```
Comment (mcjcloud) [inline x/schema/keeper/msg_server_schema.go:73]: @joshLong145 Have you made a `GetWhatIsByDID` method?

File: x/schema/keeper/msg_server_schema.go [hunk 4]
```diff
@@ -69,6 +68,39 @@ func (k msgServer) CreateSchema(goCtx context.Context, msg *types.MsgCreateSchem
...
+	if !foundSchemas {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "No Schemas found under same creator as message creator.")
+	}
+
+	var schemaWI types.WhatIs
+	var foundSchemaWI bool
+	for _, a := range schemas {
+		if a.GetDid() == msg.GetDid() {
+			schemaWI = a
+			foundSchemaWI = true
```
Comment (mcjcloud) [inline x/schema/keeper/msg_server_schema.go:83]: You can also add `break` here

File: x/schema/keeper/msg_server_schema.go [hunk 5]
```diff
@@ -69,6 +68,39 @@ func (k msgServer) CreateSchema(goCtx context.Context, msg *types.MsgCreateSchem
...
+		}
+	}
+
+	if !foundSchemaWI {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "No Schema with same creator as message creator found.")
+	}
+
+	//If already deactivated, do nothing.
+	//Responsibility of caller to check if isActive beforehand
+	if schemaWI.GetIsActive() {
```
Comment (mcjcloud) [inline x/schema/keeper/msg_server_schema.go:94]: Great optimization 👍

File: x/schema/keeper/msg_server_schema_test.go [hunk 6]
```diff
@@ -7,6 +7,21 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// func ToString(w types.WhatIs) string {
```
Comment (mcjcloud) [inline x/schema/keeper/msg_server_schema_test.go]: nit: remove commented code

**Review by mcjcloud — CHANGES_REQUESTED** [review]
Great work, just requested a few changes.

File: proto/schema/v1/genesis.proto (hunk 2 above)
Comment (prdlk) [reply proto/schema/v1/genesis.proto]: +1

File: x/schema/keeper/msg_server_schema.go (hunk 3 above)
Comment (prdlk) [reply x/schema/keeper/msg_server_schema.go:73]: @mcjcloud its fine, parsing and querying by did on chain is more unnecessary compute/gas, keep it this way and we can utilize did for network level resolution

File: x/schema/keeper/msg_server_schema.go (hunk 4 above)
Comment (prdlk) [reply x/schema/keeper/msg_server_schema.go:83]: @andym1125 yes hit a break

File: x/schema/keeper/msg_server_schema.go (hunk 5 above)
Comment (prdlk) [reply x/schema/keeper/msg_server_schema.go:94]: Good shit

**Review by prdlk — APPROVED** [review]

File: proto/schema/v1/genesis.proto (hunk 2 above)
Comment (andym1125) [reply proto/schema/v1/genesis.proto]: Changed

File: x/schema/keeper/msg_server_schema.go (hunk 4 above)
Comment (andym1125) [reply x/schema/keeper/msg_server_schema.go:83]: Changed

File: x/schema/keeper/msg_server_schema.go (hunk 5 above)
Comment (andym1125) [reply x/schema/keeper/msg_server_schema.go:94]: Thanks!

File: x/schema/keeper/msg_server_schema_test.go (hunk 6 above)
Comment (andym1125) [reply x/schema/keeper/msg_server_schema_test.go]: Changed

**Pull request closed without merging**